Relaxation of fixed binary variable raises AssertionError

```Julia
julia> using JuMP

julia> model = Model();

julia> @variable(model, x, Bin); @variable(model, y, Int);

julia> @objective(model, Max, y);

julia> @constraint(model, x + y <= 10);

julia> undo_relax = relax_integrality(model);

julia> undo_relax()

julia> fix(x, 1);

julia> println(model)
Max y
Subject to
 x + y ≤ 10.0
 x = 1.0
 y integer
 x binary


julia> undo_relax = relax_integrality(model);
ERROR: AssertionError: !(_moi_is_fixed(backend, v))
Stacktrace:
 [1] _moi_set_lower_bound(::MathOptInterface.Utilities.CachingOptimizer{MathOptInterface.AbstractOptimizer,MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.Model{Float64}}}, ::VariableRef, ::Float64) at /home/remi/.julia/packages/JuMP/qhoVb/src/variables.jl:385
 [2] set_lower_bound(::VariableRef, ::Float64) at /home/remi/.julia/packages/JuMP/qhoVb/src/variables.jl:376
 [3] relax_integrality(::Model) at /home/remi/.julia/packages/JuMP/qhoVb/src/variables.jl:1097
 [4] top-level scope at none:0

julia> println(model)
Max y
Subject to
 x + y ≤ 10.0
 x = 1.0
 y integer


julia> undo_relax = relax_integrality(model);

julia> println(model)
Max y
Subject to
 x + y ≤ 10.0
 x = 1.0
```

I would add to the end of 
https://github.com/jump-dev/JuMP.jl/blob/6c1a3bbe57a17b8a2b60fea6268d656b3b7bc815/test/variable.jl#L727-L777
```Julia
fix(x, 1)
unrelax = relax_integrality(model)
@test !is_binary(x)
@test is_fixed(x)
unrelax()
@test is_binary(x)
@test is_fixed(x)

fix(x, 0)
unrelax = relax_integrality(model)
@test !is_binary(x)
@test is_fixed(x)
unrelax()
@test is_binary(x)
@test is_fixed(x)

fix(a, 1)
unrelax = relax_integrality(model)
@test !is_integer(a)
@test is_fixed(a)
unrelax()
@test is_integer(a)
@test is_fixed(a)
```

--- a/src/variables.jl
+++ b/src/variables.jl
@@ -1097,6 +1097,10 @@ function relax_integrality(model::Model)
             if !info.has_fix
                 set_lower_bound(v, max(0.0, info.lower_bound))
                 set_upper_bound(v, min(1.0, info.upper_bound))
+            else
+                unfix(v)
+                set_lower_bound(v, max(0.0, info.fixed_value))
+                set_upper_bound(v, min(1.0, info.fixed_value))
             end
         end
     end
@@ -1117,6 +1121,10 @@ function relax_integrality(model::Model)
                     else
                         delete_upper_bound(v)
                     end
+                else
+                    delete_lower_bound(v)
+                    delete_upper_bound(v)
+                    fix(v, info.fixed_value)
                 end
             end
         end

--- a/src/variables.jl
+++ b/src/variables.jl
@@ -1094,8 +1094,10 @@ function relax_integrality(model::Model)
             unset_integer(v)
         elseif info.binary
             unset_binary(v)
-            set_lower_bound(v, max(0.0, info.lower_bound))
-            set_upper_bound(v, min(1.0, info.upper_bound))
+            if !info.has_fix
+                set_lower_bound(v, max(0.0, info.lower_bound))
+                set_upper_bound(v, min(1.0, info.upper_bound))
+            end
         end
     end
     function unrelax()
@@ -1104,15 +1106,17 @@ function relax_integrality(model::Model)
                 set_integer(v)
             elseif info.binary
                 set_binary(v)
-                if info.has_lb
-                    set_lower_bound(v, info.lower_bound)
-                else
-                    delete_lower_bound(v)
-                end
-                if info.has_ub
-                    set_upper_bound(v, info.upper_bound)
-                else
-                    delete_upper_bound(v)
+                if !info.has_fix
+                    if info.has_lb
+                        set_lower_bound(v, info.lower_bound)
+                    else
+                        delete_lower_bound(v)
+                    end
+                    if info.has_ub
+                        set_upper_bound(v, info.upper_bound)
+                    else
+                        delete_upper_bound(v)
+                    end
                 end
             end
         end

--- a/src/variables.jl
+++ b/src/variables.jl
@@ -1097,7 +1097,7 @@ function relax_integrality(model::Model)
             if !info.has_fix
                 set_lower_bound(v, max(0.0, info.lower_bound))
                 set_upper_bound(v, min(1.0, info.upper_bound))
-            else
+            elseif info.fixed_value < 0 || info.fixed_value > 1
                 unfix(v)
                 set_lower_bound(v, max(0.0, info.fixed_value))
                 set_upper_bound(v, min(1.0, info.fixed_value))
@@ -1121,7 +1121,7 @@ function relax_integrality(model::Model)
                     else
                         delete_upper_bound(v)
                     end
-                else
+                elseif info.fixed_value < 0 || info.fixed_value > 1
                     delete_lower_bound(v)
                     delete_upper_bound(v)
                     fix(v, info.fixed_value)

--- a/src/variables.jl
+++ b/src/variables.jl
@@ -1098,9 +1098,8 @@ function relax_integrality(model::Model)
                 set_lower_bound(v, max(0.0, info.lower_bound))
                 set_upper_bound(v, min(1.0, info.upper_bound))
             elseif info.fixed_value < 0 || info.fixed_value > 1
-                unfix(v)
-                set_lower_bound(v, max(0.0, info.fixed_value))
-                set_upper_bound(v, min(1.0, info.fixed_value))
+                error("The model has no valid relaxation: binary variable " *
+                      "fixed out of bounds.")
             end
         end
     end
@@ -1121,10 +1120,6 @@ function relax_integrality(model::Model)
                     else
                         delete_upper_bound(v)
                     end
-                elseif info.fixed_value < 0 || info.fixed_value > 1
-                    delete_lower_bound(v)
-                    delete_upper_bound(v)
-                    fix(v, info.fixed_value)
                 end
             end
         end

--- a/test/variable.jl
+++ b/test/variable.jl
@@ -774,6 +774,30 @@ function test_Model_relax_integrality(::Any, ::Any)
     @test is_integer(b)
     @test lower_bound(b) == -1.0
     @test upper_bound(b) == 2.0
+
+    fix(x, 1)
+    unrelax = relax_integrality(model)
+    @test !is_binary(x)
+    @test is_fixed(x)
+    unrelax()
+    @test is_binary(x)
+    @test is_fixed(x)
+
+    fix(x, 0)
+    unrelax = relax_integrality(model)
+    @test !is_binary(x)
+    @test is_fixed(x)
+    unrelax()
+    @test is_binary(x)
+    @test is_fixed(x)
+
+    fix(a, 1)
+    unrelax = relax_integrality(model)
+    @test !is_integer(a)
+    @test is_fixed(a)
+    unrelax()
+    @test is_integer(a)
+    @test is_fixed(a)
 end
 
 function test_Model_relax_integrality_error_cases(::Any, ::Any)

--- a/test/variable.jl
+++ b/test/variable.jl
@@ -814,6 +814,20 @@ function test_Model_relax_integrality_error_cases(::Any, ::Any)
     err = ErrorException("Support for relaxing semi-integer constraints " *
                          "is not yet implemented.")
     @test_throws err relax_integrality(model)
+
+    model = Model()
+    @variable(model, x, Bin)
+    fix(x, 2)
+    err = ErrorException("The model has no valid relaxation: binary variable " *
+                         "fixed out of bounds.")
+    @test_throws err relax_integrality(model)
+
+    model = Model()
+    @variable(model, x, Bin)
+    fix(x, -1)
+    err = ErrorException("The model has no valid relaxation: binary variable " *
+                         "fixed out of bounds.")
+    @test_throws err relax_integrality(model)
 end
 
 function runtests()

